File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * @param {ListNode} list1
 * @param {ListNode} list2
 * @return {ListNode}
 */



var mergeTwoLists = function(list1, list2) {
  class Bruh {
    constructor(val, next) {
      this.val = (val===undefined ? 0 : val)
      this.next = (next===undefined ? null : next)
    }
  }
  
  // idk
  
  const resList = new Bruh(0, null);
  let resTail = resList;
  
  let curr1 = list1;
  let curr2 = list2;
  
  while (curr1 && curr2) {
    if (curr1.val < curr2.val) {
      resTail.next = curr1;
      curr1 = curr1.next;
      resTail = resTail.next;      
    } else if (curr1.val >= curr2.val) {
      resTail.next = curr2;
      curr2 = curr2.next;
      resTail = resTail.next;
    }
  }
  
  if (curr1) {
    resTail.next = curr1;
  } else if (curr2) {
    resTail.next = curr2;
  }
  
  return resList.next;
};

File: 21-merge-two-sorted-lists/21-merge-two-sorted-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def mergeTwoLists(self, list1: Optional[ListNode], list2: Optional[ListNode]) -> Optional[ListNode]:
        """
        """
        dummy = ListNode(0)
        curd = dummy
        cur1, cur2 = list1, list2
        while cur1 and cur2:
            if cur1.val < cur2.val:
                curd.next = cur1
                curd = curd.next
                cur1 = cur1.next
            else:
                curd.next = cur2
                curd = curd.next
                cur2 = cur2.next
                
        if cur1:
            curd.next = cur1
        elif cur2:
            curd.next = cur2
        
        return dummy.next